otx/cli/utils/multi_gpu.py

```diff
@@ -0,0 +1,240 @@
+import torch.multiprocessing as mp
+
+from otx.api.configuration import ConfigurableParameters
+
+logger = logging.getLogger(__name__)
+
+
+def get_gpu_ids(gpus: str) -> List[int]:
+    """Get proper GPU indices form `--gpu` arguments.
+
+    Given `--gpus` argument, exclude inappropriate indices and transform to list of int format.
+
+    Args:
+        gpus (str): GPU indices to use. Format should be Comma-separated indices.
+
+    Returns:
+        List[int]:
+            list including proper GPU indices.
+    """
+    num_available_gpu = torch.cuda.device_count()
+    gpu_ids = []
+    for gpu_id in gpus.split(","):
+        if not gpu_id.isnumeric():
+            raise RuntimeError("--gpus argument should be numbers separated by ','.")
+        gpu_ids.append(int(gpu_id))
+
+    wrong_gpus = []
+    for gpu_idx in gpu_ids:
+        if gpu_idx >= num_available_gpu:
+            wrong_gpus.append(gpu_idx)
+
+    for wrong_gpu in wrong_gpus:
+        gpu_ids.remove(wrong_gpu)
+
+    if wrong_gpus:
+        logger.warning(f"Wrong gpu indeces are excluded. {','.join([str(val) for val in gpu_ids])} GPU will be used.")
+
+    return gpu_ids
+
+
+def set_arguments_to_argv(key: str, value: Optional[str] = None, after_params: bool = False):
+    """Add arguments at proper position in `sys.argv`.
+
+    Args:
+        key (str): arguement key.
+        value (str or None): argument value.
+        after_params (bool): whether argument should be after `param` or not.
+    """
+    if key in sys.argv:
+        if value is not None:
+            sys.argv[sys.argv.index(key) + 1] = value
+    else:
+        if not after_params and "params" in sys.argv:
+            sys.argv.insert(sys.argv.index("params"), key)
+            if value is not None:
+                sys.argv.insert(sys.argv.index("params"), value)
+        else:
+            if after_params and "params" not in sys.argv:
+                sys.argv.append("params")
+            if value is not None:
+                sys.argv.extend([key, value])
+            else:
+                sys.argv.append(key)
+
+
+class MultiGPUManager:
+    """Class to manage multi GPU training.
+
+    Args:
+        train_func (Callable): model training function.
+        gpu_ids (str): GPU indices to use. Format should be Comma-separated indices.
+        multi_gpu_port (str): port for communication between multi GPU processes.
+    """
+
+    def __init__(self, train_func: Callable, gpu_ids: str, multi_gpu_port: str):
+        self._train_func = train_func
+        self._gpu_ids = get_gpu_ids(gpu_ids)
+        self._multi_gpu_port = multi_gpu_port
+        self._main_pid = os.getpid()
+        self._processes: Optional[List[mp.Process]] = None
+
+    def is_available(self) -> bool:
+        """Check multi GPU training is available.
+
+        Returns:
+            bool:
+                whether multi GPU training is available.
+        """
+        return len(self._gpu_ids) > 1
+
+    def setup_multi_gpu_train(
```

Comment:
Similar with above, `setup_train` is okay for me.

Reply:
I think this function name can convey meaning better.